activiti-cloud-modeling-service/activiti-cloud-services-modeling/activiti-cloud-services-modeling-service/src/main/java/org/activiti/cloud/services/modeling/service/ProjectServiceImpl.java

```diff
@@ -293,27 +294,21 @@ private void convertZipElementToModelObject(ZipStream.ZipStreamEntry zipEntry, @
      */
     @Override
     public Project importProject(MultipartFile file, @Nullable String name) throws IOException {
-        ProjectHolder projectHolder = new ProjectHolder();
-
-        ZipStream.of(file)
-                .forEach(zipEntry -> createFileContentFromZipEntry(zipEntry)
-                        .ifPresent(fileContent -> convertZipElementToModelObject(zipEntry, name, fileContent, projectHolder)));
-
-        Project createdProject = projectHolder.getProjectMetadata().map(this::createProject)
-                .orElseThrow(() -> new ImportProjectException("No valid project entry found to import: " + file.getOriginalFilename()));
-
-        projectHolder.getModelJsonFiles().forEach(modelJsonFile -> {
-            importJSONModelFiles(projectHolder, createdProject, modelJsonFile);
-        });
-
-        projectHolder.getModelContentFiles().forEach(modelXmlFile ->
-                importXMLModelFiles(projectHolder, createdProject, modelXmlFile.getModelType(), modelXmlFile.getFileContent()));
-
-        Map<Model, FileContent> createdProcesses = createXMLModelFiles(projectHolder, createdProject);
-        createdProcesses.keySet().forEach(model -> updateModelProcessImported(projectHolder, model, createdProcesses.get(model)));
+        return importModelsFromProjectHolder(getProjectHolderFromZipStream(ZipStream.of(file), name), file.getOriginalFilename());
```

Comment:
Does this ZipStream need to be closed?

Reply:
I don't think so. The zipstream is used below inside a try-with-resources statement

Reply:
From what I remember the good practice to follow is: "who opens the stream should take care of closing it". In this case, `ZipStream.of(file)` is opening the stream with `file.getInputStream()`, but it cannot close it, as the stream is used later on. So maybe it's better to completely remove `ZipStream.of(MultipartFile file)` and change it to:
```
try (InputStream inputStream = file.getInputStream()) {
            return importModelsFromProjectHolder(getProjectHolderFromZipStream(ZipStream.of(inputStream), name));
        }
```